junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/ParameterResolutionUtilsTests.java

```diff
@@ -53,38 +50,40 @@
 
 	private final JupiterConfiguration configuration = mock(JupiterConfiguration.class);
 
-	private MutableExtensionRegistry extensionRegistry = MutableExtensionRegistry.createRegistryWithDefaultExtensions(
+	private final MutableExtensionRegistry extensionRegistry = MutableExtensionRegistry.createRegistryWithDefaultExtensions(
 		configuration);
 
 	@Test
-	void constructorInjection() {
-		register(new StringParameterResolver(), new NumberParameterResolver());
+	void resolveConstructorArguments() {
+		register(new StringParameterResolver());
 
-		Class<ConstructorInjectionTestCase> outerClass = ConstructorInjectionTestCase.class;
-		Constructor<ConstructorInjectionTestCase> constructor = ReflectionUtils.getDeclaredConstructor(outerClass);
-		ConstructorInjectionTestCase outer = newInvoker().invoke(constructor, Optional.empty(), extensionContext,
-			extensionRegistry, passthroughInterceptor());
+		Class<ConstructorInjectionTestCase> topLevelClass = ConstructorInjectionTestCase.class;
+		Object[] arguments = resolveConstructorParameters(topLevelClass, null);
+
+		assertThat(arguments).hasSize(1).containsExactly(ENIGMA);
+	}
+
+	@Test
+	void resolveNestedConstructorArguments() {
+		register(new NumberParameterResolver());
 
-		assertNotNull(outer);
-		assertEquals(ENIGMA, outer.str);
+		Class<ConstructorInjectionTestCase> outerClass = ConstructorInjectionTestCase.class;
+		ConstructorInjectionTestCase outer = ReflectionUtils.newInstance(outerClass, "str");
 
 		Class<ConstructorInjectionTestCase.NestedTestCase> innerClass = ConstructorInjectionTestCase.NestedTestCase.class;
-		Constructor<ConstructorInjectionTestCase.NestedTestCase> innerConstructor = ReflectionUtils.getDeclaredConstructor(
-			innerClass);
-		ConstructorInjectionTestCase.NestedTestCase inner = newInvoker().invoke(innerConstructor, Optional.of(outer),
-			extensionContext, extensionRegistry, passthroughInterceptor());
+		Object[] arguments = resolveConstructorParameters(innerClass, outer);
 
-		assertNotNull(inner);
-		assertEquals(42, inner.num);
+		assertThat(arguments).hasSize(2).containsExactly(outer, 42);
```

Comment:
```suggestion
		assertThat(arguments).containsExactly(outer, 42);
```

`hasSize()` is already covered by `containsExactly()`